crates/turbopack-dev/js/src/runtime.js

```diff
@@ -276,6 +279,7 @@ function instantiateModule(id, source) {
   /** @type {Module} */
   const module = {
     exports: {},
+    error: undefined,
```

Comment:
We shouldn't expose this on the module object itself. Let's store in a hash map instead.

Reply:
hmm... That would make receiving the error really expensive.

```diff
-  return getOrInstantiateModuleFromParent(id, sourceModule).exports;
+  const module = getOrInstantiateModuleFromParent(id, sourceModule);
+  const internalState = internalState.get(module); // <- expensive
+  if (internalState.error) throw module.error;
+  return module.exports;
```

So I don't think it's worth the isolation here.

Also webpack is exposing this via `module.error` too, so it's probably even more compatible when exposing it.